Sources/PackageModel/PackageReference.swift

```diff
@@ -9,63 +9,108 @@
 */
 
 import Basics
+import Foundation
 import TSCBasic
 import TSCUtility
 
 /// A package reference.
 ///
 /// This represents a reference to a package containing its identity and location.
-public struct PackageReference: Codable {
+public struct PackageReference: Encodable {
     /// The kind of package reference.
-    public enum Kind: String, Codable {
+    public enum Kind: Equatable {
         /// A root package.
-        case root
+        case root(AbsolutePath)
 
         /// A non-root local package.
-        case local
-
-        /// A remote package.
-        case remote
+        case fileSystem(AbsolutePath)
+
+        /// A local source package.
+        case localSourceControl(AbsolutePath)
+
+        /// A remote source package.
+        case remoteSourceControl(Foundation.URL)
+
+        // FIXME: we should not need this
+        //@available(*, deprecated)
+        public var locationString: String {
+            switch self {
+            case .root(let path):
+                return path.pathString
+            case .fileSystem(let path):
+                return path.pathString
+            case .localSourceControl(let path):
+                return path.pathString
+            case .remoteSourceControl(let url):
+                return url.absoluteString
+            }
+        }
+
+        // FIXME: ideally this would not be required and we can check on the enum directly
+        public var isRoot: Bool {
+            if case .root = self {
+                return true
+            } else {
+                return false
+            }
+        }
     }
 
     /// The identity of the package.
     public let identity: PackageIdentity
 
     /// The name of the package, if available.
+    // FIXME: we should not need this
     public var name: String
 
-    /// The path of the package.
+    /// The location of the package.
     ///
     /// This could be a remote repository, local repository or local package.
-    public let location: String
+    // FIXME: we should not need this
+    //@available(*, deprecated)
```

Comment:
after landing this PR, there will be follow up to clean up call sites that still depend on location string. there are very few now, mostly for display so we are getting there

Reply:
Great, this will do a lot to clarify the meaning of the code.